src/sentry/static/sentry/app/components/events/eventEntries.tsx

```diff
@@ -45,39 +65,174 @@ type Props = {
   event: Event;
   project: Project;
   location: Location;
-
+  api: Client;
   group?: Group;
   className?: string;
 } & typeof defaultProps;
 
-class EventEntries extends React.Component<Props> {
+type State = {
+  isLoading: boolean;
+  proGuardErrors: ProGuardErrors;
+};
+
+class EventEntries extends React.Component<Props, State> {
   static defaultProps = defaultProps;
 
+  state: State = {
+    isLoading: true,
+    proGuardErrors: [],
+  };
+
   componentDidMount() {
+    this.checkProGuardError();
+    this.recordIssueError();
+  }
+
+  shouldComponentUpdate(nextProps: Props, nextState: State) {
+    const {event, showExampleCommit} = this.props;
+
+    return (
+      (event && nextProps.event && event.id !== nextProps.event.id) ||
+      showExampleCommit !== nextProps.showExampleCommit ||
+      nextState.isLoading !== this.state.isLoading
+    );
+  }
+
+  async fetchDebugFile(query: string): Promise<Array<DebugFile>> {
+    const {api, organization, project} = this.props;
+    try {
+      const debugFiles = await api.requestPromise(
+        `/projects/${organization.slug}/${project.slug}/files/dsyms/`,
+        {
+          method: 'GET',
+          query: {
+            query,
+            file_formats: organization.features?.includes('android-mappings')
+              ? ['breakpad', 'macho', 'elf', 'pe', 'pdb', 'sourcebundle']
+              : undefined,
+          },
+        }
+      );
+      return debugFiles;
+    } catch (error) {
+      Sentry.captureException(error);
+      // do nothing, the UI will not display extra error details
+      return [];
+    }
+  }
+
+  isDataMinified(str: string | null) {
+    if (!str) {
+      return false;
+    }
+
+    return !![...str.matchAll(MATCH_MINIFIED_DATA_REGEX)].length;
+  }
+
+  hasThreadOrExceptionMinifiedFrameData(event: Event, bestThread?: Thread) {
+    if (!bestThread) {
+      const exceptionValues: Array<ExceptionValue> =
+        event.entries?.find(e => e.type === EntryType.EXCEPTION)?.data?.values ?? [];
+
+      return !!exceptionValues.find(exceptionValue =>
+        exceptionValue.stacktrace?.frames?.find(frame =>
+          this.isDataMinified(frame.module)
+        )
+      );
+    }
+
+    const threadExceptionValues = getThreadException(event, bestThread)?.values;
+
+    return !!(threadExceptionValues
+      ? threadExceptionValues.find(threadExceptionValue =>
+          threadExceptionValue.stacktrace?.frames?.find(frame =>
+            this.isDataMinified(frame.module)
+          )
+        )
+      : bestThread?.stacktrace?.frames?.find(frame => this.isDataMinified(frame.module)));
+  }
+
+  async checkProGuardError() {
     const {event} = this.props;
 
-    if (!event || !event.errors || !(event.errors.length > 0)) {
+    if (!event) {
```

Comment:
It must be platform=java for ProGuard to kick in, so we can also guard for that:

```suggestion
    if (!event || event.platform !== 'java') {
```

Reply:
yes I can add this to another part of the code. This code was already there and it's not related to my changes